CodeEdit/Commands/CommandPaletteView.swift

```diff
@@ -0,0 +1,285 @@
+
+            Divider()
+            VStack(spacing: 0) {
+                List(commandsList, selection: $state.selected) { command in
+                    // swiftlint:disable multiple_closures_with_trailing_closure
+                    Button(action: { onCommandClick(command: command) }) {
+                        VStack {
+                            SearchResultLabel(labelName: command.title, textToMatch: state.commandQuery,
+                                              fontColor: textColor(command: command))
+                                .padding(EdgeInsets.init(top: 0, leading: 8, bottom: 0, trailing: 0))
+                        }.frame(maxWidth: .infinity, maxHeight: 15, alignment: .leading)
+                    }.frame(maxWidth: .infinity, maxHeight: 15, alignment: .leading)
+                        .listRowInsets(EdgeInsets.init(top: 0, leading: 0, bottom: 0, trailing: 0))
+                        .padding(EdgeInsets.init(top: 5, leading: 5, bottom: 5, trailing: 0))
+                        .buttonStyle(.borderless)
+                        .background(self.selectedItem == command ?
+                                           RoundedRectangle(cornerRadius: 5, style: .continuous)
+                            .fill(Color(red: 0, green: 0.38, blue: 0.816, opacity: 0.85)) :
+                                            RoundedRectangle(cornerRadius: 5, style: .continuous)
+                            .fill(Color.clear))
+                        .onHover(perform: { _ in self.selectedItem = command })
+                    }.listStyle(SidebarListStyle())
+
+            }
+        }
+        .background(EffectView(.sidebar, blendingMode: .behindWindow))
+        .foregroundColor(.gray)
+        .edgesIgnoringSafeArea(.vertical)
+        .frame(minWidth: 600,
+           minHeight: self.state.isShowingCommandsList ? 400 : 28,
+           maxHeight: self.state.isShowingCommandsList ? .infinity : 28)
+    }
+}
+
+struct CommandPaletteView_Previews: PreviewProvider {
+    static var previews: some View {
+        CommandPaletteView(
+            state: .init(),
+            closePalette: {}
+        )
+    }
+}
+
+class ActionAwareInputView: NSTextView, NSTextFieldDelegate {
+
+    var onDown: ((NSEvent) -> Bool)?
+    var onTextChange: ((String) -> Void)?
+
+    func control(_ control: NSControl, textView: NSTextView, doCommandBy commandSelector: Selector) -> Bool {
+        return true
+    }
+
+    override var acceptsFirstResponder: Bool { return true }
+
+    override public func keyDown(with event: NSEvent) {
+        if onDown!(event) {
+            // We don't want to pass event down the pipe if it was handled.
+            // By handled I mean its keycode was used for something else than typing
+            return
+        }
+
+        super.keyDown(with: event)
+    }
+
+    override public func didChangeText() {
+        onTextChange?(self.string)
+    }
+
+}
+
+public struct SearchResultLabel: NSViewRepresentable {
```

Comment:
Please add proper documentation to `public` classes, structs, methods and variables/constants.

A guide about code documentation can be found [here](https://nshipster.com/swift-documentation/).